atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/TableMappingService.java

```diff
@@ -27,5 +27,5 @@
     TableReference getMappedTableName(TableReference tableRef) throws TableMappingNotFoundException;
     <T> Map<TableReference, T> mapToShortTableNames(Map<TableReference, T> tableMap)
             throws TableMappingNotFoundException;
-    Set<TableReference> mapToFullTableNames(Set<TableReference> tableNames);
+    Map<TableReference, TableReference> generateMapToFullTableNames(Set<TableReference> tableNames);
```

Comment:
Only the large internal product is expected to use this API. As long as we change that case it is fine. We dont have any set guarantees around our public API, but generally try to avoid breaks. I do agree the change without the API break is not elegant and we can go ahead with this as is if you can confirm the large internal product will not have a problem due to the change.